### PR TITLE
fix(kv-cache): fill quantized attention masks with dtype finfo.min, not leastNormalMagnitude

### PR DESCRIPTION
**Pull request by mimeding**
This is the offered fix for #133. The three boolean mask paths in `quantizedScaledDotProductAttention` filled excluded logits with `Float.leastNormalMagnitude`, the smallest positive normal Float, so masked logits were effectively zero and could retain softmax mass. They now use `scores.dtype.finfo!.min`, matching Python mlx-lm's `finfo(scores.dtype).min`; this is finite rather than `-inf` because an entirely masked row of infinities produces a NaN softmax row.

The affected code is not reached by vanilla one-token decode with no mask. It is reachable for multi-token quantized-cache steps, including speculative-decode verification.

Each fill scalar is constructed explicitly in `scores.dtype` before `MLX.where`, avoiding promotion of bf16 or fp16 scores by a float32 scalar. The additive floating-point mask branches are intentionally unchanged: caller-provided additive masks already encode their intended bias values, so no fill happens there.

Focused KV-cache tests now cover causal-mask parity with `MLXFast.scaledDotProductAttention`, boolean-array-mask parity (fp16 and bf16) and zero masked-position weight, a fully masked row remaining finite (that test documents the finite-fill semantics rather than the constant change), and unchanged additive-mask routing. The fixtures use tiny tensors and a real `QuantizedKVCache`; they require no model downloads.

One adjacent observation from reviewing for other occurrences, left out of this PR to keep it verifiable: `Gemma3nText.swift:599` builds its sliding-window exclusion with the same `Float.leastNormalMagnitude` constant (for fp16 masks it underflows to exactly +0, so positions beyond the window keep an additive bias of 0), and `:854` uses it as an epsilon that also underflows to 0 in fp16. Happy to file that separately or fold it in here if you prefer.

Happy for you to take this over or adjust the constant if leastNormalMagnitude was intentional for a Metal-side reason.

### CODE REVIEW

**Comment by jjang-ai** [comment]
### QA — the tests genuinely catch the bug, and no family shifts

**Sabotage check first**, because a mask test that passes either way proves
nothing. Merged this branch onto main, then reverted ONLY
`KVCache.swift` (keeping the new tests) so `leastNormalMagnitude` was back in
all 3 fills:

```
✘ testQuantizedAttentionCausalMaskMatchesReference        — 2 issues
✘ testQuantizedAttentionBooleanArrayMaskMatchesReference  — float16:  2 issues
✘ testQuantizedAttentionBooleanArrayMaskMatchesReference  — bfloat16: 2 issues
✘ Test run failed with 6 issues
```

With the fill restored to `scores.dtype.finfo!.min`, all pass:

```
✔ testQuantizedAttentionCausalMaskMatchesReference
✔ testQuantizedAttentionFullyMaskedRowIsFinite
✔ testQuantizedAttentionBooleanArrayMaskMatchesReference (2 cases: float16, bfloat16)
```

So the coverage is real, and it catches the bug in both half-precision dtypes —
which is where the explicit `dtype:` construction matters, since a float32
scalar would have promoted the scores.

`testQuantizedAttentionFullyMaskedRowIsFinite` is the right third test: it pins
why this is `finfo.min` and not `-inf`, so nobody "tidies" it later into a NaN
softmax row.

**No family moved.** Full KVCache suite passes, and on the merged build:

| family | check | result |
|---|---|---|
| LFM2.5-VL-3B-JANG_4M | full variating pattern | `diff vs main -> IDENTICAL` (18 lines) |
| Bonsai-27b-Ternary | growing-chat cache | `HIT tier=paged matched=18/53` — passed |

The variating run crosses reasoning ON/OFF, text↔image↔text-after-image, tools
offered/called, image AND tools in one turn, two images in one turn, a
constant-head grow control (`HIT disk 22/38, 57% reused`) and turn 1 replayed
verbatim (`HIT disk 26/26, 100% reused`).

Identical output is the expected result and worth stating plainly: as you note,
the affected fills are not reached by vanilla one-token decode with no mask, so
these runs demonstrate the change is inert on the default path — the behavioural
proof is the sabotage above, not the bench.

One process note for anyone reproducing: don't `git checkout <pr> -- KVCache.swift`
onto today's main. That file has since gained `HybridPoolQuantizedSegment`
plumbing, so taking the PR's whole file drops it and the build fails with
`cannot find 'HybridPoolQuantizedSegment' in scope` — a stale binary then makes
a bench diff look clean. Merge the branch instead.
